### PR TITLE
[full-ci] Forward port stable 5.0

### PR DESCRIPTION
**Pull request by kulmann**
## Description
Forward porting `stable-5.0` and manually adding missing changelog items from the `stable-4.0` branch in order to bring  the changelog (will be built in CI) up to date on `master`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)
- [x] Maintenance

### CODE REVIEW

File: changelog/4.0.5_2023-12-21/add-trach-bin-cli.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Enhancement: Add cli commands for trash-binq
```
Comment (mmattel) [inline changelog/4.0.5_2023-12-21/add-trach-bin-cli.md:1]:
```suggestion
Enhancement: Add cli commands for trash-bin
```
typo

**Review by mmattel — DISMISSED** [review]
I have to believe you that all is done correctly 😅

**Comment by kulmann** [comment]
> I have to believe you that all is done correctly 😅 

In that case it's not a good idea to approve the PR.

File: changelog/5.0.0-rc.4_2024-02-07/fix-signed-url-expiry.md
Comment (phil-davis) [inline changelog/5.0.0-rc.4_2024-02-07/fix-signed-url-expiry.md:1]: This changelog is also already in https://github.com/owncloud/ocis/blob/master/changelog/unreleased/fix-signed-url-expiry.md

Comment (kulmann) [reply changelog/5.0.0-rc.4_2024-02-07/fix-signed-url-expiry.md:1]: Thanks for spotting that! Removed it from `unreleased`.

**Review by 2403905 — APPROVED** [review]

**Pull request merged**